src/v/json/allocator.h

```diff
@@ -0,0 +1,48 @@
+#include "vlog.h"
+
+#include <fmt/format.h>
+#include <rapidjson/allocators.h>
+namespace json {
+
+class bad_alloc_with_msg final : public std::bad_alloc {
+public:
+    explicit bad_alloc_with_msg(std::string msg)
+      : _msg(std::move(msg)) {}
+
+    const char* what() const noexcept override { return _msg.c_str(); }
+
+private:
+    std::string _msg;
+};
+
+class throwing_allocator {
+public:
+    void* Malloc(size_t size) {
+        void* res = _rp_allocator.Malloc(size);
+        if (!res && (0 != size)) {
+            throw bad_alloc_with_msg(
```

Comment:
this is fine, but you can remove all allocations if you log this and return a static string. just an idea